### ISSUE TITLE
allYourBase still used in twisted.persisted.styles

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5193]
|[<img alt="dustin's avatar" src="https://avatars.githubusercontent.com/u/28673?s=50" width="50" height="50">](https://github.com/dustin)| @dustin reported|
|-|-|
|Trac ID|trac#5193|
|Type|enhancement|
|Created|2011-07-09 23:08:01Z|
|Branch|https://github.com/twisted/twisted/tree/aybabtu-warnings-5193|

I see

```
  /home/dustin/code/buildbot/t/buildbot/sand27/lib/python2.7/site-packages/twisted/persisted/styles.py:160: DeprecationWarning: twisted.python.reflect.allYourBase was deprecated in Twisted 11.0.0; please use inspect.getmro instead
```

in my logs quite a lot. Obviously it's harmless, but should be fixed.

Attachments:

* [bug5193-1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ecc/eccf8e455eca5d64c9ad752f8dc1db144dbc1daa/429471926225123b7e92b4ce7fbe6c91b0a438fb.patch) (3007 bytes) - added by dustin on 2011-07-09 23:43:40Z - 5193-1.patch
* [bug5193-2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ecc/eccf8e455eca5d64c9ad752f8dc1db144dbc1daa/2d8ca7673a57d38d9e6bd92b765fbcfc25a8692a.patch) (64 bytes) - added by dustin on 2011-08-15 19:29:24Z - bug5193-2.patch
* [bug5193-2.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ecc/eccf8e455eca5d64c9ad752f8dc1db144dbc1daa/1c2a6656960fdee98396cd1b52c00abc926032fc.patch) (2348 bytes) - added by dustin on 2011-08-15 22:07:02Z - bug5193-2.patch
* [bug5193-3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ecc/eccf8e455eca5d64c9ad752f8dc1db144dbc1daa/e0b71bfe5997de768b9f65532c6552845a226ee6.patch) (2674 bytes) - added by dustin on 2011-08-17 18:59:34Z - bug5193-3.patch

<details><summary>Searchable metadata</summary>

```
trac-id__5193 5193
type__enhancement enhancement
reporter__dustin dustin
priority__normal normal
milestone__ 
branch__branches_aybabtu_warnings_5193 branches/aybabtu-warnings-5193
branch_author__dustin dustin
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1310252881000000 1310252881000000
changetime__1313612833000000 1313612833000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dustin's avatar" src="https://avatars.githubusercontent.com/u/28673?s=50" width="50" height="50">](https://github.com/dustin)<a name="note_1"></a>|@dustin commented|
|-|-|

As [#4928](https://github.com/twisted/twisted/issues/4928) mentions, this is one of the few places that the second argument to allYourBase is used, and as such getmro isn't a drop-in replacement for it.  [#3843](https://github.com/twisted/twisted/issues/3843) recommends deprecating styles.Versioned, but that hasn't happened yet and anyway, it doesn't make sense for one part of twisted to use a function deprecated in another part of Twisted.

**Comment by twisted-trac** [comment]
|[<img alt="dustin's avatar" src="https://avatars.githubusercontent.com/u/28673?s=50" width="50" height="50">](https://github.com/dustin)<a name="note_2"></a>|@dustin removed owner|
|-|-|

The included tests pass with the old (allYourBase) implementation of _aybabtu, as well.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [32444]) Branching to 'aybabtu-warnings-5193'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [32445]) Apply bug5193-1.patch 

refs [#5193](https://github.com/twisted/twisted/issues/5193)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @dustin|
|-|-|

Thanks.  Some simple points:

  1. There's extra space in the list definition in `_aybabtu`, should be `[c, Versioned]`
  1. The new test method should be `test_aybabtu` rather than `testAybabtu`
  1. The new test method's docstring is pretty weak.  Don't write docstrings that say "Test foo".  It's a test method, what else would it be doing? :)  Write docstrings that describe desired behavior, to give the reader of the test method implementation some context.
  1. This test could really be 4 different tests, probably.
  1. `AybabtuTests` could use a docstring too (this is a more appropriate place to write "Test foo", though some extra description of what foo does in general is also nice).

**Comment by twisted-trac** [comment]
|[<img alt="dustin's avatar" src="https://avatars.githubusercontent.com/u/28673?s=50" width="50" height="50">](https://github.com/dustin)<a name="note_6"></a>|@dustin removed owner|
|-|-|

This (bug5193-2.patch) should fix all of the above points.  The docstrings feel a little bit like "x++; // add one to x" to me, but I think they are what you want?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set owner to @dustin|
|-|-|

Replying to [dustin](#note_6):
> This (bug5193-2.patch) should fix all of the above points.  The docstrings feel a little bit like "x++; // add one to x" to me, but I think they are what you want?

Thanks. The patch only contains the following though:

```
# On branch bug5193
nothing to commit (working directory clean)
```

**Comment by twisted-trac** [comment]
|[<img alt="dustin's avatar" src="https://avatars.githubusercontent.com/u/28673?s=50" width="50" height="50">](https://github.com/dustin)<a name="note_8"></a>|@dustin removed owner|
|-|-|

Er, sorry, hg has rotted my brain.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [32459]) Apply bug5193-2.2.patch 

refs [#5193](https://github.com/twisted/twisted/issues/5193)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks!  Some style points:

Docstrings should be formatted like:
```
def foo():
    """
    bar
    """
```

Test methods should be like:
```
def test_fooBarBaz(self):
    ...
```

Since there is a built in type _set_, the test method docstrings should probably talk about _list_ instead, as that is the actual return type.

These are minor points, I'll make the changes and merge the branch.  Thanks again.

**Comment by twisted-trac** [comment]
|[<img alt="dustin's avatar" src="https://avatars.githubusercontent.com/u/28673?s=50" width="50" height="50">](https://github.com/dustin)<a name="note_11"></a>|@dustin commented|
|-|-|

Actually, that raises a good point - the order of the results is not guaranteed by ```_ababtu```, and the tests should reflect that -- probably by testing ```assertEqual(sorted(...), sorted([expected])```.  I'll make another patch - one moment.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

(In [32460]) Fix minor style issues

refs [#5193](https://github.com/twisted/twisted/issues/5193)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [32461]) Merge aybabtu-warnings-5193

Author: dustin
Reviewer: exarkun
Fixes: [#5193](https://github.com/twisted/twisted/issues/5193)

Remove the use of the deprecated `twisted.python.reflect.allYourBase` from
`twisted.persisted.styles`.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

> Actually, that raises a good point - the order of the results is not guaranteed by _ababtu

Clearly trac is not good for realtime interactions. :/  However, the ordering isn't non-deterministic, so I don't have a problem with the tests asserting a particular order.
